include/seastar/core/smp.hh

```diff
@@ -334,8 +343,11 @@ class smp : public std::enable_shared_from_this<smp> {
     using returns_future = is_future<std::invoke_result_t<Func>>;
     template <typename Func>
     using returns_void = std::is_same<std::invoke_result_t<Func>, void>;
+private:
+    void setup_prefaulter(seastar::resource::hwloc::internal::topology_holder& topo, seastar::memory::internal::numa_layout layout);
 public:
-    explicit smp(alien::instance& alien) : _alien(alien) {}
+    explicit smp(alien::instance& alien);
+    ~smp();
```

Comment:
nit: It doesn't look like this change belongs to this patch

Reply:
Probably leftover from try { patch } catch { do something else }

Reply:
It's necessary since memory_prefaulter is forward declared, so its destructor is not available here. I'll enhance the changelog.